src/components/TodoList.tsx

```diff
@@ -0,0 +1,373 @@
+        })
+        .catch(() => {
+          setTodosToBeEdited(null);
+          setError?.(ErrorMessage.CantDelete);
+        });
+    }
+  };
+
+  const handleTodoNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (editingTitleTodo !== null) {
+      setEditingTitleTodo({
+        id: editingTitleTodo.id,
+        userId: editingTitleTodo.userId,
+        completed: editingTitleTodo.completed,
+        title: event.target.value,
+      });
+    }
+  };
+
+  const handleFinishTodoNameChange = (todo: Todo) => {
+    if (editingTitleTodo) {
+      if (editingTitleTodo.title === todo.title) {
+        setEditingTitleTodo(null);
+      } else if (!editingTitleTodo.title) {
+        handleDeletion(todo.id);
+      } else {
+        setTodosToBeEdited([todo.id]);
+        patchTodo(editingTitleTodo.id, { title: editingTitleTodo.title })
+          .then(() => {
+            setTodosToBeEdited([]);
+            const todoList = [...todos];
+
+            const todoIndex = todoList
+              .findIndex(arrTodo => arrTodo.id === todo.id);
+
+            todoList[todoIndex] = {
+              id: todo.id,
+              userId: todo.userId,
+              title: editingTitleTodo.title,
+              completed: todo.completed,
+            };
+
+            setEditingTitleTodo(null);
+            setTodos(todoList);
+          })
+          .catch(() => {
+            setEditingTitleTodo(null);
+            setError?.(ErrorMessage.CantUpdate);
+          });
+      }
+    }
+  };
+
+  const handleOnKeyUp = (
+    event: React.KeyboardEvent<HTMLElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      setEditingTitleTodo(null);
+    }
+
+    // if (event.key === 'Enter') {
+    //   event.preventDefault();
+    //   handleFinishTodoNameChange(todo);
+    // }
+  };
+
+  const handleSubmitEditedTodo = (
+    event: React.FormEvent<HTMLFormElement>, todo: Todo,
+  ) => {
+    event.preventDefault();
+    handleFinishTodoNameChange(todo);
+  };
+
+  return (
+    <>
+      <header className="todoapp__header">
+        {/* this buttons is active only if there are some active todos */}
+        <button
+          type="button"
+          className={cn({
+            'todoapp__toggle-all': true,
+            active: !todos.find(todo => !todo.completed),
+          })}
+          aria-label="Toggle all"
+          onClick={handleMassActiveToggle}
+        />
+
+        <form>
+          <input
+            type="text"
+            className="todoapp__new-todo"
+            placeholder="What needs to be done?"
+            value={todoTitle}
+            onChange={(event) => setTodoTitle(event.target.value)}
+            onKeyDown={handleSubmitNewTodo}
+            disabled={processing}
+          />
+        </form>
+      </header>
+
+      <section className="todoapp__main">
+        {filteredTodos?.map(todo => (
+          <div
+            className={cn({
+              todo: true,
+              completed: todo.completed,
```

Comment:
destructure todo

Reply:
is changing `onClick={() => handleActiveToggle(todo)}` to `onClick={() => handleActiveToggle({ id, title, userId, completed )}` acceptable in this case?